src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationParsingTests.cs

```diff
@@ -7272,6 +7272,44 @@ public void TestMethodDeclarationNullValidation()
             }
         }
 
+        [Fact]
+        public void TestMethodDeclarationNullValidation_SingleExclamation()
+        {
+            UsingStatement(@"void M(string name!) { }", options: TestOptions.RegularPreview,
+                    // (1,19): error CS1003: Syntax error, '!!' expected
```

Comment:
>'!!' expected [](http://example.com/codeflow?start=59&length=13)

Are we testing this case with trailing and leading trivia on the `!` token? Perhaps:

```csharp
void M(string name
/*comment1*/!/*comment2*/);
```

Reply:
Added the test in ae0c6bf